src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123C58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_SetEquipmentData.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123D50.s")
+extern u8 sActionModelGroups[];
+
+s32 Player_ActionToModelGroup(Player* player, s32 actionParam) {
+    s32 modelGroup = sActionModelGroups[actionParam];
+
+    if ((modelGroup == 2) && func_801234B0(player)) {
+        return 1;
+    }
+    return modelGroup;
+}
+
+extern u8 D_801BFF90[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF90[PLAYER_FORM_MAX] = {
+    PLAYER_BOOTS_FIERCE_DEITY, // PLAYER_FORM_FIERCE_DEITY
+    PLAYER_BOOTS_GORON,        // PLAYER_FORM_GORON
+    PLAYER_BOOTS_ZORA_LAND,    // PLAYER_FORM_ZORA
+    PLAYER_BOOTS_DEKU,         // PLAYER_FORM_DEKU
+    PLAYER_BOOTS_HYLIAN,       // PLAYER_FORM_HUMAN
+};
+#endif
+
+extern u8 D_801BFF98[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF98[PLAYER_FORM_MAX] = {
+    1, // PLAYER_FORM_FIERCE_DEITY
+    3, // PLAYER_FORM_GORON
+    2, // PLAYER_FORM_ZORA
+    0, // PLAYER_FORM_DEKU
+    1, // PLAYER_FORM_HUMAN
+};
+#endif
+
+typedef struct {
+    /* 0x00 */ u8 flag;
+    /* 0x02 */ u16 textId;
+} TextTriggerEntry; // size = 0x04
+
+extern TextTriggerEntry D_801BFFA0[];
+#if 0
+TextTriggerEntry D_801BFFA0[] = {
+    { 1, 0x26FC },
+    { 2, 0x26FD },
+    { 0, 0 },
+    { 2, 0x26FD },
+};
+#endif
+
+extern Gfx** sPlayerDListGroups[];
+
+void func_801239AC(Player* player) {
+    if (player->stateFlags1 & PLAYER_STATE1_400000) {
+        if ((player->heldItemActionParam < 0) || (player->heldItemActionParam == player->itemActionParam)) {
+            if (!Player_HoldsTwoHandedWeapon(player)) {
+                if (!func_801234B0(player)) {
+                    D_801F59E0 = player->transformation * 2;
+                    player->rightHandType = 8;
+                    player->rightHandDLists = &sPlayerDListGroups[8][D_801F59E0];
+
+                    if (player->sheathType == 14) {
+                        player->sheathType = 12;
+                    } else if (player->sheathType == 15) {
+                        player->sheathType = 13;
+                    }
+
+                    player->sheathDLists = &sPlayerDListGroups[player->sheathType][D_801F59E0];
+                    player->modelAnimType = 2;
+                    player->heldItemActionParam = -1;
+                }
+            }
+        }
+    }
+}
+
+void Player_SetModels(Player* player, s32 modelGroup) {
+    u8* aux;
+
+    D_801F59E0 = player->transformation * 2;
+    player->leftHandType = gPlayerModelTypes[modelGroup][1];
+    player->rightHandType = gPlayerModelTypes[modelGroup][2];
+    player->sheathType = gPlayerModelTypes[modelGroup][3];
+
+    if (player->sheathType == 14) {
+        if (gSaveContext.equips.buttonItems[CUR_FORM][0] == ITEM_NONE) {
+            player->sheathType = 15;
+        }
+    }
+
+    aux = gPlayerModelTypes[modelGroup];
```

Comment:
`aux` seems like a weird temp name. Can't you just use something like `playerModelTypes` idk